src/components/pseudonym/repo/pseudonym.repo.js

```diff
@@ -1,25 +1,27 @@
 const Pseudonym = require('../../../services/pseudonym/model');
-const { GeneralError } = require('../../../errors');
+const { deleteManyResult } = require('../../helper/repo.helper');
+const { validateObjectId } = require('../../helper/uc.helper');
 
+/**
+ * Return pseudonyms for userId
+ * @param userId
+ */
 const getPseudonymsForUser = async (userId) => {
+	validateObjectId(userId);
 	return Pseudonym.find({ userId }).lean().exec();
 };
 
-const deletePseudonyms = async (pseudonymIds) => {
-	const deleteResult = await Pseudonym.deleteMany({
-		_id: {
-			$in: pseudonymIds,
-		},
-	})
-		.lean()
-		.exec();
-	if (deleteResult.n !== deleteResult.ok || deleteResult.ok !== pseudonymIds.length) {
-		throw new GeneralError('db error during deleting pseudonyms');
-	}
-	return pseudonymIds;
+/**
+ * Removes all pseudonyms for userId
+ * @param {String|ObjectId} userId
+ */
+const deletePseudonymsForUser = async (userId) => {
+	validateObjectId(userId);
+	const deleteResult = await Pseudonym.deleteMany({ userId }).lean().exec();
```

Comment:
extract the filter here and above and use the same filter in delete/get
`const byUserFilter = (userId) => {return {userId}};`?
This should ensure not changing one filter only in the future... ;)

Reply:
I don't get the reason why is it necessary here.